src/components/TaskInput.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+type TaskInputProps = {
+  placeholder: string
+  value: string
+  handleTaskAdd: (e: {code: string;}) => void
+  handleTaskChange: (event: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
/nit naming convention for event handlers in react is usually something like `on<X>` such as `onTaskAdd`

Also, you could make these slightly tidier for the parent component by providing some abstraction on top of the native event handling, i.e:
```
taskName: string
onTaskAdd: (taskName: string) => void
onChangeTaskName: (taskName: string) => void
```